packages/casl-ability/src/builder.js

```diff
@@ -40,20 +51,25 @@ export class AbilityBuilder {
     };
   }
 
-  constructor() {
+  constructor({ subjectName = getSubjectName } = {}) {
     this.rules = [];
+    this[PRIVATE_FIELD] = {
+      subjectName
+    };
   }
 
   can(actions, subject, conditionsOrFields, conditions) {
     if (!isStringOrNonEmptyArray(actions)) {
       throw new TypeError('AbilityBuilder#can expects the first parameter to be an action or array of actions');
     }
 
-    if (!isStringOrNonEmptyArray(subject)) {
-      throw new TypeError('AbilityBuilder#can expects the second argument to be a subject name or array of subject names');
+    const subjectName = toNames(subject, this[PRIVATE_FIELD].subjectName);
```

Comment:
lets rewrite this a bit shorter `const subjectName = [].concat(subject).map(this.subjectName)`

Reply:
this will allow to remove `toNames` as well